### PR TITLE
En 4003 sc results interceptors refactor

### PR DESCRIPTION
**Pull request by iulianpascalau**
refactored interceptor for unsigned tx (smart contract result) to use the new interceptors architecture

### CODE REVIEW

File: process/dataValidators/nilTxValidator.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	return &nilTxValidator{}, nil
+}
+
+// IsTxValidForProcessing is a nil implementation that will return true
+func (ntv *nilTxValidator) IsTxValidForProcessing(interceptedTx process.TxValidatorHandler) bool {
+	return true
+}
+
+// NumRejectedTxs will return number of rejected transaction
+func (ntv *nilTxValidator) NumRejectedTxs() uint64 {
```
Comment (bogdan-rosianu) [inline process/dataValidators/nilTxValidator.go]: not tested

File: process/interceptors/factory/interceptedDataFactory_test.go [hunk 2]
```diff
@@ -177,6 +179,23 @@ func TestInterceptedDataFactory_CreateInterceptedTxShouldWork(t *testing.T) {
...
+
+	marshalizer := &mock.MarshalizerMock{}
+	emptyTx := &smartContractResult.SmartContractResult{}
+	emptyTxBuff, _ := marshalizer.Marshal(emptyTx)
+
+	idf, _ := factory.NewInterceptedDataFactory(createMockArgument(), factory.InterceptedUnsignedTx)
+
+	instance, err := idf.Create(emptyTxBuff)
+
+	assert.NotNil(t, instance)
```
Comment (bogdan-rosianu) [inline process/interceptors/factory/interceptedDataFactory_test.go]: remove empty line and add empty line before type assertion

File: process/unsigned/interceptedUnsignedTransaction.go [hunk 3]
```diff
@@ -109,53 +112,67 @@ func (inUTx *InterceptedUnsignedTransaction) integrity() error {
...
-func (inUTx *InterceptedUnsignedTransaction) verifyIfNotarized(uTxBuff []byte) error {
-	// TODO: implement this for flood protection purposes
-	return nil
+// Nonce returns the transaction nonce
+func (inUTx *InterceptedUnsignedTransaction) Nonce() uint64 {
+	return inUTx.uTx.Nonce
+}
+
+// SenderAddress returns the transaction sender address
+func (inUTx *InterceptedUnsignedTransaction) SenderAddress() state.AddressContainer {
```
Comment (bogdan-rosianu) [inline process/unsigned/interceptedUnsignedTransaction.go]: not tested

File: process/unsigned/interceptedUnsignedTransaction.go [hunk 4]
```diff
@@ -14,15 +14,16 @@ import (
...
-	isAddressedToOtherShards bool
+	uTx          *smartContractResult.SmartContractResult
+	marshalizer  marshal.Marshalizer
+	hasher       hashing.Hasher
+	addrConv     state.AddressConverter
+	coordinator  sharding.Coordinator
+	hash         []byte
+	rcvShard     uint32
+	sndShard     uint32
+	isForMyShard bool
```
Comment (bogdan-rosianu) [inline process/unsigned/interceptedUnsignedTransaction.go]: I suggest renaming it to _isForCurrentShard_ but it's up to you

File: process/factory/shard/interceptorsContainerFactory.go [hunk 5]
```diff
@@ -309,15 +308,32 @@ func (icf *interceptorsContainerFactory) generateUnsignedTxsInterceptors() ([]st
...
+	txProcessor, err := processor.NewTxInterceptorProcessor(argProcessor)
+	if err != nil {
+		return nil, err
+	}
 
-	interceptor, err := unsigned.NewUnsignedTxInterceptor(
+	txFactory, err := interceptorFactory.NewInterceptedDataFactory(
+		icf.argInterceptorFactory,
+		interceptorFactory.InterceptedUnsignedTx,
+	)
```
Comment (AdoAdoAdo) [inline process/factory/shard/interceptorsContainerFactory.go]: you need to check the error for creating the interceptor data factory as well

File: process/dataValidators/nilTxValidator.go [hunk 6]
```diff
@@ -0,0 +1,32 @@
+package dataValidators
+
+import (
+	"github.com/ElrondNetwork/elrond-go/process"
+)
+
+// nilTxValidator represents a tx handler validator that doesn't check the validity of provided txHandler
+type nilTxValidator struct {
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply process/dataValidators/nilTxValidator.go]: ok, will move to the mock folder

File: process/dataValidators/nilTxValidator.go (hunk 1 above)
Comment (iulianpascalau) [reply process/dataValidators/nilTxValidator.go]: Will remove all tests as it becomes a mock

File: process/interceptors/factory/interceptedDataFactory_test.go (hunk 2 above)
Comment (iulianpascalau) [reply process/interceptors/factory/interceptedDataFactory_test.go]: made modifications to conform to arrange-test-assert structure

File: process/unsigned/interceptedUnsignedTransaction.go (hunk 3 above)
Comment (iulianpascalau) [reply process/unsigned/interceptedUnsignedTransaction.go]: add test

File: process/unsigned/interceptedUnsignedTransaction.go (hunk 4 above)
Comment (iulianpascalau) [reply process/unsigned/interceptedUnsignedTransaction.go]: Done

File: process/dataValidators/nilTxValidator.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply process/dataValidators/nilTxValidator.go]: Done

File: process/factory/shard/interceptorsContainerFactory.go (hunk 5 above)
Comment (iulianpascalau) [reply process/factory/shard/interceptorsContainerFactory.go]: Good catch, found another one. Fixed.

File: process/transaction/interceptedTransaction.go [hunk 7]
```diff
@@ -125,7 +125,7 @@ func (inTx *InterceptedTransaction) processFields(txBuff []byte) error {
 
 	isMyShardRecv := inTx.rcvShard == inTx.coordinator.SelfId()
```
Comment (bogdan-rosianu) [inline process/transaction/interceptedTransaction.go]: maybe rename these 2 variables also

Comment (iulianpascalau) [reply process/transaction/interceptedTransaction.go]: Done

**Pull request merged**